opentelemetry-api/opentelemetry/trace/__init__.py

```diff
@@ -73,6 +73,11 @@ class Tracer:
     and controlling spans' lifecycles.
     """
 
+    """Constant used to represent the current span being used as a parent.
+    This is the default behavior when creating spans.
+    """
+    CURRENT_SPAN = object()
```

Comment:
Consider `_CURRENT_SPAN` if this is meant to be used inside this file.

Reply:
One more point: This will not typecheck (CURRENT_SPAN does not have type `Span`). We should use [`typing.NewType`](https://docs.python.org/3/library/typing.html#typing.NewType) to create a dummy type for this object and then accept that type in the functions that support CURRENT_SPAN (in a `typing.Union`).

Reply:
> Actually we need to expose it, as API implementations will need to compare against it.

Correct. My initial thought had been to hide it but we have to expose it ;)

And yes, we can use `typing.NewType` for this. But I'm actually thinking we could simply try to use the Java approach here, by having a **specific** default/no-op `Span` that will effectively act as such dummy. Will update the PR ;)